applications/MappingApplication/custom_mappers/nearest_neighbor_mapper_iga.h

```diff
@@ -0,0 +1,305 @@
+
+    void CalculateAll(MatrixType& rLocalMappingMatrix,
+                      EquationIdVectorType& rOriginIds,
+                      EquationIdVectorType& rDestinationIds,
+                      MapperLocalSystem::PairingStatus& rPairingStatus) const override;
+
+    CoordinatesArrayType& Coordinates() const override
+    {
+        KRATOS_DEBUG_ERROR_IF_NOT(mpNode) << "Members are not initialized!" << std::endl;
+        return mpNode->Coordinates();
+    }
+
+    MapperLocalSystemUniquePointer Create(NodePointerType pNode) const override
+    {
+        return Kratos::make_unique<NearestNeighborLocalSystemIGA>(pNode);
+    }
+
+    void PairingInfo(std::ostream& rOStream, const int EchoLevel) const override;
+
+    void SetPairingStatusForPrinting() override;
+
+private:
+    NodePointerType mpNode;
+
+};
+
+/// Nearest Neighbor Mapper for IGA-FEM simulations 
+/** This class implements the Nearest Neighbor Mapping technique for IGA-FEM partitioned simulations.
+* Each node on the destination side (FEM side) gets assigned is's closest integration point (neighbor) on the other side of the interface (IGA side).
+* Once the nearest integration point is found, the shape function values of the integration point are used to interpolate the values from the IGA side to the FEM side.
+* For information about the available echo_levels and the JSON default-parameters
+* look into the class description of the MapperCommunicator
+*/
+template<class TSparseSpace, class TDenseSpace, class TMapperBackend>
+class KRATOS_API(MAPPING_APPLICATION) NearestNeighborMapperIGA
+    : public InterpolativeMapperBase<TSparseSpace, TDenseSpace, TMapperBackend>
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    ///@}
+    ///@name Pointer Definitions
+    /// Pointer definition of NearestNeighborMapperIGA
+    KRATOS_CLASS_POINTER_DEFINITION(NearestNeighborMapperIGA);
+
+    typedef InterpolativeMapperBase<TSparseSpace, TDenseSpace, TMapperBackend> BaseType;
+    typedef typename BaseType::MapperUniquePointerType MapperUniquePointerType;
+    typedef typename BaseType::MapperInterfaceInfoUniquePointerType MapperInterfaceInfoUniquePointerType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    // Default constructor, needed for registration
+    NearestNeighborMapperIGA(ModelPart& rModelPartOrigin,
+                             ModelPart& rModelPartDestination)
+                             : BaseType(rModelPartOrigin, rModelPartDestination) {}
+
+    NearestNeighborMapperIGA(ModelPart& rModelPartOrigin,
+                             ModelPart& rModelPartDestination,
+                             Parameters JsonParameters)
+                             : BaseType(rModelPartOrigin,
+                                       rModelPartDestination,
+                                       JsonParameters)
+    {
+        KRATOS_TRY;
+
+        KRATOS_ERROR_IF(!JsonParameters.Has("is_origin_iga") || JsonParameters["is_origin_iga"].GetBool() != true)
```

Comment:
```suggestion
        KRATOS_ERROR_IF(!JsonParameters.Has("is_origin_iga") || !JsonParameters["is_origin_iga"].GetBool())
```

comparing bool to true or false is a punishable programming crime :laughing: